src/queries.ts

```diff
@@ -308,12 +314,31 @@ export function useSubmissionCreate() {
   return mutation;
 }
 
-export function useSubmissionSchema() {
-  return useQuery({
-    queryKey: submissionKeys.submissionSchema(),
+export interface SubmissionSchema {
+  schema: SchemaDefinition;
+  goldEcosystemTree: GoldEcosystemTreeNode;
+}
+function submissionSchemaQueryOptions(queryClient: QueryClient) {
+  // The `queryOptions` function is to help with type inference
+  // https://tanstack.com/query/latest/docs/framework/react/typescript#typing-query-options
+  return queryOptions({
+    queryKey: schemaKeys.submissionSchema(),
     // These two files are used in conjunction with each other so make the two requests in parallel
     // and return an object bundling the results together.
     queryFn: async () => {
+      const cachedData = queryClient.getQueryData<SubmissionSchema>(
+        schemaKeys.submissionSchema(),
+      );
+      if (cachedData !== undefined) {
+        // We have a copy of the schema in the cache. Check to see if it is still the latest version
+        const versionInfo = await nmdcServerClient.getVersionInfo();
+        // If the cached schema is the same version as the current server version, return the cached
+        // schema
+        if (cachedData.schema.version === versionInfo.nmdc_submission_schema) {
+          return cachedData;
+        }
+      }
```

Comment:
Non-deal-breaker nit-pick: I moved the comments to above the `if` block to contrast it with the `// We don't have a ...` block that follows the `if` block.

```suggestion
      // If we have a cached schema and it's the same version as the server
      // would give us if we were to fetch it now, return that cached schema.
      if (cachedData !== undefined) {
        const versionInfo = await nmdcServerClient.getVersionInfo();
        if (cachedData.schema.version === versionInfo.nmdc_submission_schema) {
          return cachedData;
        }
      }
```